alvr/audio/src/linux.rs

```diff
@@ -71,6 +71,13 @@ pub fn play_microphone_loop_pipewire(
             average_buffer_frames_count,
         )
         .ok();
+
+        // if we end up here then no consumer is currently connected to the output,
+        // so discard audio packets to not cause a buildup
+
+        if let Err(ConnectionError::Other(_)) = receiver.recv(Duration::from_millis(500)) {
+            break;
+        };
```

Comment:
Two things. If you match but don't need to the value, you should prefer `matches!()` instead of `let`. Second thing, this code seem wrong. If you break here then you exit pipewire completely. what I would do is: remove the added code, and before `let stream_audio` add:
```rust
let is_streaming = pw_stream_state_arc.try_lock().map(|s| *s != StreamState::Streaming).unwrap_or(false) {
if !is_streaming {
    receiver.recv(Duration::from_millis(500)).ok();
    continue;
}
```
To note is that we release the lock before blocking in recv.

Reply:
> If you match but don't need to the value, you should prefer matches!() instead of let

Was looking for that bud I'm bad at rust's syntactic sugar

> If you break here then you exit pipewire completely

That's intentional, the windows code also just exits completely there, so I mirrored that.

The audio loop is called here:
https://github.com/alvr-org/ALVR/blob/master/alvr/server_core/src/connection.rs#L830
Which calls `receive_samples_loop` without any restart mechanism
And `receive_samples_loop` just exits with an error if a category other error is encountered
https://github.com/alvr-org/ALVR/blob/master/alvr/audio/src/lib.rs#L446

And moving it beforehand doesn't really help, because the receive loop still needs to know when the audio consumer gets disconnected, so why duplicate the checking logic

Reply:
> https://github.com/alvr-org/ALVR/blob/master/alvr/server_core/src/connection.rs#L830

Uhm this line is `#[cfg(not(target_os = "linux"))]`

We don't want to exit the pipewire thread, because we want to keep it alive in case a consumer connects. 

> And moving it beforehand doesn't really help, because the receive loop still needs to know when the audio consumer gets disconnected, so why duplicate the checking logic

Ah I think I get it... yeah. The "ConnectionError:: Other" means the channel got disconnected, while no more buffers is "Try again". So you code works. Inside `receive_samples_loop()` we create a buffer, but we define it as empty so it doesn't allocate, so issues here.

So just switch to `matches!()` and it's done.